core/templates/pages/collection-editor-page/navbar/collection-editor-navbar-breadcrumb.component.spec.ts

```diff
@@ -0,0 +1,133 @@
+    tags: ['mock tag'],
+    playthrough: null,
+    category: 'Collection Category',
+    version: 0,
+    schemaVersion: 1,
+    nodes: []
+  };
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+      declarations: [
+        CollectionEditorNavbarBreadcrumbComponent
+      ],
+      providers: [
+        CollectionEditorStateService,
+        CollectionEditorRoutingService,
+        FocusManagerService
+      ],
+      schemas: [
+        NO_ERRORS_SCHEMA
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(
+      CollectionEditorNavbarBreadcrumbComponent);
+    component = fixture.componentInstance;
+    collectionEditorStateService = TestBed.inject(CollectionEditorStateService);
+    collectionEditorStateService = (collectionEditorStateService as unknown) as
+      jasmine.SpyObj<CollectionEditorStateService>;
+    collectionEditorRoutingService = TestBed.inject(
+      CollectionEditorRoutingService);
+    collectionEditorRoutingService = (
+      (collectionEditorRoutingService as unknown) as
+        jasmine.SpyObj<CollectionEditorRoutingService>);
+    focusManagerService = TestBed.inject(FocusManagerService);
+    focusManagerService = (focusManagerService as unknown) as
+      jasmine.SpyObj<FocusManagerService>;
+  });
+
+  beforeEach(() => {
+    mockCollection = new Collection(
+      collectionData.id, collectionData.title, collectionData.objective,
+      collectionData.languageCode, collectionData.tags,
+      collectionData.playthrough, collectionData.category,
+      collectionData.version, collectionData.schemaVersion,
+      collectionData.nodes);
+  });
+
+  it('should load the component on opening collection editor page by clicking' +
+    'on create collection', () => {
+    spyOn(collectionEditorStateService, 'getCollection').and.returnValue(
+      mockCollection);
+    spyOn(collectionEditorRoutingService, 'getActiveTabName').and.returnValue(
+      activeTab);
+
+    component.ngOnInit();
+    expect(component.collection).toBe(mockCollection);
+    expect(component.activeTabName).toBe(activeTab);
+  });
+
+  it('should get the current tab name in readable form', () => {
+    spyOn(collectionEditorStateService, 'getCollection').and.returnValue(
+      mockCollection);
+    spyOn(collectionEditorRoutingService, 'getActiveTabName').and.returnValue(
+      activeTab);
+
+    component.ngOnInit();
+    expect(component.getCurrentTabName()).toBe('Edit');
+  });
+
+  it('should change the active tab to settings when clicked on' +
+    'collection title if the title is empty', fakeAsync(() => {
+    spyOn(collectionEditorStateService, 'getCollection').and.returnValue(
+      mockCollection);
+    spyOn(collectionEditorRoutingService, 'getActiveTabName').and.returnValue(
+      activeTab);
+    spyOn(focusManagerService, 'setFocus');
+
+    component.ngOnInit();
```

Comment:
Is it possible to place the `component.ngOnInit();` in the `beforeEach` block?

Reply:
done